test/specs/jsCodegen/string.tests.ts

```diff
@@ -0,0 +1,52 @@
+import * as chai from 'chai';
+import { ReturnType } from 'fontoxpath';
+import { sanitizeString } from 'fontoxpath/jsCodegen/escapeJavaScriptString';
+import * as slimdom from 'slimdom';
+import jsonMlMapper from 'test-helpers/jsonMlMapper';
+import evaluateXPathWithJsCodegen from './evaluateXPathWithJsCodegen';
+
+describe('string tests', () => {
```

Comment:
Can you add a test verifying evaluating a query like `child::element()` is not compiling to return a node by accident?

Reply:
Sure, I'll do that.